### PR TITLE
Initial buildout

### PR DESCRIPTION
**Pull request by pjhartzell**
**Description:**
Initial build of project after boilerplate creation.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Examples have been updated to reflect changes, if applicable
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,26 +11,47 @@
...
   - [eo](https://github.com/stac-extensions/eo)
   - [item-assets](https://github.com/stac-extensions/item-assets)
+  - [mgrs](https://github.com/stac-extensions/mgrs)
   - [proj](https://github.com/stac-extensions/projection)
   - [raster](https://github.com/stac-extensions/raster)
   - [scientific](https://github.com/stac-extensions/scientific)
+  - [view](https://github.com/stac-extensions/view)
 - Extra fields:
-  - `hls:tile-id`: Tile identifier using the Military Grid Reference System (MGRS)
+  - `hls:product`: "HLSL30" (Landsat) or "HLSS30" (Sentinel) product
```
Comment (gadomski) [inline README.md:20]: What are your thoughts on implementing a custom STAC extension, ala https://github.com/stactools-packages/noaa-mrms-qpe/tree/main/extension? Too much work for too little payoff?

File: src/stactools/hls/commands.py [hunk 2]
```diff
@@ -13,43 +17,119 @@ def create_hls_command(cli: Group) -> Command:
...
+        default=False,
+        help="Check that all granule asset COGs exist",
     )
-    @click.argument("destination")
-    def create_collection_command(destination: str) -> None:
-        """Creates a STAC Collection
-
+    @click.option(
+        "-a",
+        "--antimeridian_strategy",
```
Comment (gadomski) [inline src/stactools/hls/commands.py]:
```suggestion
        "--antimeridian-strategy",
```

File: src/stactools/hls/commands.py [hunk 3]
```diff
@@ -13,43 +17,119 @@ def create_hls_command(cli: Group) -> Command:
...
+
+        for href in hrefs:
+            item = stac.create_item(
+                href, check_existence=check_existence, antimeridian_strategy=strategy
+            )
+            collection.add_item(item)
+        collection.update_extent_from_items()
+
+        collection.catalog_type = CatalogType.SELF_CONTAINED
+        collection.make_all_asset_hrefs_relative
```
Comment (gadomski) [inline src/stactools/hls/commands.py]:
```suggestion
        collection.make_all_asset_hrefs_relative()
```

File: src/stactools/hls/fragments/__init__.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+from stactools.hls.utils import band_name_from_href, product_from_href
+
+
+class STACFragments:
+    """Class for accessing collection and asset data."""
+
+    def __init__(self) -> None:
+        self.assets = self._load("assets.json")
+        for key in self.assets.keys():
+            self.assets[key]["type"] = MediaType.COG
```
Comment (gadomski) [inline src/stactools/hls/fragments/__init__.py:18]: Is `assets` meant to be accessed directly, or only through the `self.asset()` method? If it is not supposed to be used directly, this attribute should be made private (`self._assets`).

File: src/stactools/hls/stac.py [hunk 5]
```diff
@@ -1,117 +1,154 @@
...
+    read_href_modifier: Optional[ReadHrefModifier] = None,
+    check_existence: bool = False,
+    antimeridian_strategy: Strategy = Strategy.SPLIT,
+) -> Item:
+    """Creates a STAC Item for an HLS granule.
 
-    See `Collection<https://pystac.readthedocs.io/en/latest/api.html#collection>`_.
+    Args:
+        cog_href (str): HREF to one of the EO COG files in the granule.
+        read_href_modifier (Optional[ReadHrefModifier], optional): An optional
```
Comment (gadomski) [inline src/stactools/hls/stac.py]:
```suggestion
        read_href_modifier (ReadHrefModifier, optional): An optional
```

**Review by gadomski — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (pjhartzell) [reply README.md:20]: First reaction is, yes, it seems a bit much for a single field. Not that it wouldn't be a good exercise. 

Add as a `feature request` issue?

File: src/stactools/hls/commands.py (hunk 3 above)
Comment (pjhartzell) [reply src/stactools/hls/commands.py]: Erf. Done.

File: src/stactools/hls/fragments/__init__.py (hunk 4 above)
Comment (pjhartzell) [reply src/stactools/hls/fragments/__init__.py:18]: It is accessed [directly in `create_collection`](https://github.com/stactools-packages/hls/blob/82dfa755f6a833dcb15107579346f73a4bb3496d/src/stactools/hls/stac.py#L135) for building `item_assets`.

File: src/stactools/hls/stac.py (hunk 5 above)
Comment (pjhartzell) [reply src/stactools/hls/stac.py]: Thanks. Cleaned up all other occurrences as well.

**Review by gadomski — APPROVED** [review]

**Comment by sharkinsspatial** [comment]
@pjhartzell I just stumbled on this repo while looking up another stactools repo 🎊 .  We currently generate inline STAC items for distribution with the HLS data as part of the processing pipeline using this older, poorly maintained lib https://github.com/NASA-IMPACT/hls-cmr_stac.

Not sure what your ingestion plans are for the HLS data but we can definitely consider adopting this repo for generating items in our next HLS pipeline minor release if that would streamline things for you and provide consistent items (Prior to implementing the `projection` extension we avoided directly accessing files via `rasterio` but  that is a moot point now).

**Review by pholleway — APPROVED** [review]

**Comment by gadomski** [comment]
@sharkinsspatial, @pjhartzell's on PTO for the week so I'll jump in just to say we'd love to collaborate here! Our plans are to ingest HLS into Microsoft Planetary Computer, though we intentionally keep all Microsoft-specific data/logic out of this repo.

If there were STAC metatada provided w/ HLS we would love to use it, e.g. we use USGS's Landsat STAC when possible. So much the better if its all coming from the same repo.

**Comment by sharkinsspatial** [comment]
@gadomski I'll try to get some details on when our next minor pipeline release will be but in the near term, once this PR is merged I'll start testing this package in our development environment. 

A few notes about HLS data consistency that might be helpful for your ingestion/usage.  

- All data produced prior to mid-May 2022 uses a CRS with incomplete datum information due to an issue with the underlying HDF-EOS lib used in our pipeline.  This CRS information was updated in our production system in mid-May with https://github.com/NASA-IMPACT/hls-hdf_to_cog/pull/9/files.  With older versions of `rasterio` this incomplete datum information can result in range read performance issues due to some EPSG code lookup paths.  This performance issue is fixed with https://github.com/rasterio/rasterio/releases/tag/1.3b3 but there is discussion of possible NASA reprocessing effort to update the CRS information for older data in the archive as well.

- The Landsat 9 data was included in the pipeline in our mid-May release.  The reverse processing for the Landsat 9 data from Sep 2021 through May 17 2022 will be released in mid-August.

- All the STAC items prior to mid-May have the same hemisphere EPSG issue as the Landsat stactools package https://github.com/stactools-packages/landsat/issues/12

File: src/stactools/hls/metadata.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+            cog_href (str): HREF to the COG file
+            read_href_modifier (ReadHrefModifier, optional): An
+                optional function to modify the href (e.g. to add a token to a
+                url)
+
+        Returns:
+            Metadata: A Metadata dataclass
+        """
+        read_h5_href = utils.modify_href(cog_href, read_href_modifier)
+        with rasterio.open(read_h5_href) as dataset:
```
Comment (pjhartzell) [inline src/stactools/hls/metadata.py]: This `read_h5_href` term is a copy-paste leftover. Edit to remove the h5 term, .e.g., make it `read_cog_href`.

File: README.md (hunk 1 above)
Comment (pjhartzell) [reply README.md:20]: See #2

**Comment by pjhartzell** [comment]
@sharkinsspatial Good info, definitely interested in working with you on this stactools package!

- I updated the EPSG code extraction to handle the pre-May 2022 data. The update also forces the hemisphere to be north.
- Yep, we are handling Landsat 9 data. Right now the package will handle a combination of Landsat 8 and 9 data in a single granule. But that seems unlikely. Can you confirm/deny that granules are exclusive to a single platform? 
- The XML geometries are much better than the current use of a bounding box obtained from one of the COG image bounds (#3), so that will need to be updated. I was trying to limit the number of files touched, but worth opening the XML for the geometry data.
- A note about `eo:bands` in the [hls-cmr_stac](https://github.com/NASA-IMPACT/hls-cmr_stac) repo: I see instrument specific center wavelengths and FWHM values. My understanding is that the Sentinel data is bandpassed to match the Landsat spectral response, so we are using a single set of center wavelengths and FWHM values. Is my understanding correct?

**Pull request merged**